include/aspect/simulator.h

```diff
@@ -647,8 +648,10 @@ namespace aspect
        * @param advection_field Determines whether this variable should select
        * the temperature field or a compositional field.
        */
-      void get_artificial_viscosity (Vector<float> &viscosity_per_cell,
-                                     const AdvectionField &advection_field) const;
+      template <typename T>
+      void get_artificial_viscosity (Vector<T> &viscosity_per_cell,
```

Comment:
Why make this a template? I think we should just settle on either `T=double` or `T=float`.